internal/generate/generate.go

```diff
@@ -257,3 +298,56 @@ func (c *CA) StoreAsConfigMap(ctx context.Context, k8sClient *kubernetes.Clients
 	}
 	return err
 }
+
+// StoreAsConfigMapWithKey creates or updates the CA certificate in a K8s configmap
+func (c *CA) StoreAsConfigMapWithKey(ctx context.Context, k8sClient *kubernetes.Clientset) error {
```

Comment:
Maybe I'm missing some context (I'm not familiar with the VM support yet), but I don't think we should store the CA private key in a `ConfigMap`, we should use a `Secret` instead. 

For the Hubble CA we are using a `ConfigMap` because we are only storing the public key.

Reply:
Right, will change.